internal/monitors/kubernetes/cluster/metrics/dimensions.go

```diff
@@ -243,3 +255,67 @@ func (dh *DimensionHandler) handleAddJob(job *batchv1.Job) {
 		dh.emitDimension(dimensionForPodCronJob(pod, cronJobRef.Name, cronJobRef.UID))
 	}
 }
+
+type dimensionDeduplicator struct {
```

Comment:
this seems like something that does not necessarily need to confined to this monitor. do you want to move this some place more global?

Reply:
Yes, I thought about it some more and decided to move it back into the writer's dimension client.